### DISCUSSION TITLE
Any reason why you chose the name "isDisabled" instead of the already existing "disabled"?

### DISCUSSION THREAD

**Discussion by GermanJablo** [adobe/react-spectrum#6045] [Q&A] [answered]
Considering that disabled is a property of many elements in React such as `React.ButtonHTMLAttributes<HTMLButtonElement>`, wouldn't it be better to follow the same naming convention?

It's a bit of a long story, but the question arises because this caused a bug in my code. One component ended up accepting `disabled` and `isDisabled` when in fact only 1 took effect.

**Comment by devongovett** [comment, marked as answer]
Consistency between all boolean props across all components is the reason. See our [API naming guidelines](https://github.com/adobe/react-spectrum/blob/main/specs/api/Guidelines.md) for details. We decided a long time ago to name boolean props/variables with consistent prefixes for clarity.

I would have expected TypeScript to have caught if you passed `disabled` instead of `isDisabled`. Was that not working for some reason?
